src/main/java/com/nike/cerberus/archaius/client/ArchaiusCerberusClientFactory.java

```diff
@@ -16,36 +16,90 @@
+
+    private static final String CERBERUS_REGION_SYS_PROPERTY = "cerberus.region";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ArchaiusCerberusClientFactory.class);
+
     /**
-     * Resolves the Cerberus/Cerberus URL via the {@link ArchaiusCerberusUrlResolver} and creates a new {@link CerberusClient}
-     * with the {@link DefaultCerberusCredentialsProviderChain}.
+     * Creates a new {@link CerberusClient} with the {@link DefaultCerberusCredentialsProviderChain}.
      *
      * @return Cerberus client
      */
     public static CerberusClient getClient() {
-        final ArchaiusCerberusUrlResolver archaiusUrlResolver = new ArchaiusCerberusUrlResolver();
-
         final Map<String, String> defaultHeaders = new HashMap<>();
         final String xCerberusClientHeaderValue = ClientVersion.getClientHeaderValue();
         defaultHeaders.put(ClientVersion.CERBERUS_CLIENT_HEADER, xCerberusClientHeaderValue);
 
         return CerberusClientFactory.getClient(
-                archaiusUrlResolver,
+                resolveUrl(),
                 // pass the client HTTP header value to be used in authenticate calls to Cerberus
-                new DefaultCerberusCredentialsProviderChain(archaiusUrlResolver, xCerberusClientHeaderValue),
+                new DefaultCerberusCredentialsProviderChain(resolveUrl(), getRegion(), xCerberusClientHeaderValue),
                 // pass the client header to be used in all other calls to Cerberus (e.g. read, write, etc.)
                 defaultHeaders);
     }
+
+    /**
+     * Attempts to acquire Cerberus URL from Archaius.
+     *
+     * @return Cerberus URL
+     */
+    private static String resolveUrl() {
+        final AbstractConfiguration configuration = ConfigurationManager.getConfigInstance();
+        final String envUrl = configuration.getString(CERBERUS_ADDR_ENV_PROPERTY);
+        final String sysUrl = configuration.getString(CERBERUS_ADDR_SYS_PROPERTY);
+
+        if (StringUtils.isNotBlank(envUrl) && HttpUrl.parse(envUrl) != null) {
+            return envUrl;
+        } else if (StringUtils.isNotBlank(sysUrl) && HttpUrl.parse(sysUrl) != null) {
+            return sysUrl;
+        } else {
+            LOGGER.warn("Unable to resolve the Cerberus URL.");
+            throw new IllegalArgumentException("Cerberus Address property is not configured correctly.");
+        }
+    }
+
+    /**
+     * Attempts to acquire Cerberus region from Archaius.
+     *
+     * @return Cerberus region
+     */
+    private static String getRegion() {
+        final AbstractConfiguration configuration = ConfigurationManager.getConfigInstance();
+        final String envRegion = configuration.getString(CERBERUS_REGION_ENV_PROPERTY);
+        final String sysRegion = configuration.getString(CERBERUS_REGION_SYS_PROPERTY);
+
+        if (StringUtils.isNotBlank(envRegion) && Regions.fromName(envRegion) != null) {
```

Comment:
I think if `envRegion` is null then `Regions.fromName(envRegion)` will throw `IllegalArgumentException` (Regions.java:78). What is going to happen is that if `envRegion` is null, exception will be thrown at line 96 even `sysRegion` is not null.

Reply:
`StringUtils.isNotBlank()` checks if a String is not empty (""), not null and not whitespace only.

So I think if `envRegion` is null, then the `if` statement at line 96 will short-circuit and go on to check the `sysRegion`.

Reply:
You're right. Does `Regions.fromName(envRegion)` ever return null?

Reply:
Yeah so null will never be returned, so this null check is not totally meaningful since what I really wanted this to do is to see if any exception is thrown when `Regions.fromName()` is called. Should/how can I make this clear in the code?

Reply:
I think it's fine.